File: kernel/micro/src/test/java/ink/reactor/microkernel/logger/PrintlnLogger.java
package ink.reactor.microkernel.logger;

import ink.reactor.kernel.logger.Logger;
import ink.reactor.kernel.logger.LoggerFormatter;
import org.jspecify.annotations.NonNull;

public final class PrintlnLogger implements Logger {
    private final JavaLoggerFormatter loggerFormatter = new JavaLoggerFormatter();

    @Override
    public @NonNull LoggerFormatter getLoggerFormatter() {
        return loggerFormatter;
    }

    @Override
    public void debug(final @NonNull String message) {
        System.out.println("[DEBUG]" + message);
    }

    @Override
    public void log(final @NonNull String message) {
        System.out.println("[LOG]" + message);
    }

    @Override
    public void info(final @NonNull String message) {
        System.out.println("[INFO]" + message);
    }

    @Override
    public void warn(final @NonNull String message) {
        System.out.println("[WARN]" + message);
    }

    @Override
    public void error(final @NonNull String message) {
        System.out.println("[ERROR]" + message);
    }

    @Override
    public void error(final @NonNull String message, final Throwable throwable) {
        System.err.println("[ERROR]" + message);
        throwable.printStackTrace(System.err);
    }
}

File: kernel/micro/src/test/kotlin/ink/reactor/microkernel/logger/PrintlnLogger.kt
package ink.reactor.microkernel.logger

import ink.reactor.kernel.logger.Logger

class PrintlnLogger : Logger {
    override val loggerFormatter: JavaLoggerFormatter = JavaLoggerFormatter()

    override fun debug(message: String) {
        println("[DEBUG] $message")
    }

    override fun log(message: String) {
        println("[LOG] $message")
    }

    override fun info(message: String) {
        println("[INFO] $message")
    }

    override fun warn(message: String) {
        println("[WARN] $message")
    }

    override fun error(message: String) {
        println("[ERROR] $message")
    }

    override fun error(message: String, throwable: Throwable) {
        System.err.println("[ERROR] $message")
        throwable.printStackTrace(System.err)
    }
}
